core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1097,6 +1094,15 @@ object GroupCoordinator {
     new GroupCoordinator(config.brokerId, groupConfig, offsetConfig, groupMetadataManager, heartbeatPurgatory, joinPurgatory, time)
   }
 
+  def joinError(memberId: String, error: Errors): JoinGroupResult = {
```

Comment:
Perhaps if you want to move this, we may as well put it in the `JoinGroupResult` companion object.
```scala
object JoinGroupResult {
  def apply(memberId: String, error: Errors): JoinGroupResult = {
...
  }
}
```